### DISCUSSION TITLE
How to update existing installation with external trust anchor cert?

### DISCUSSION THREAD

**Discussion by miguelvr** [linkerd/linkerd2#10425] [Help] [answered]
hey team!

how do I upgrade an existing linkerd installation to use an external trust anchor secret?

the instructions [here](https://linkerd.io/2.12/tasks/automatically-rotating-control-plane-tls-credentials/) only mention an installation from scratch

**Comment by alpeb** [comment, marked as answer]
You can upgrade linkerd as instructed in the [docs](https://linkerd.io/2.12/tasks/upgrade/#upgrading-the-control-plane), making sure the `identity.externalCA` setting is enabled, and assuming the `linkerd-identity-trust-roots` ConfigMap is provisioned externally.
Note that if you're doing this through the CLI, the flag is `--set identity.externalCA=true`.
The dataplane pods will need to be restarted in order to pick the new trust root. The transition can cause downtime, but you can avoid that by bundling together the old and new cert, similarly as described [here](https://linkerd.io/2.12/tasks/manually-rotating-control-plane-tls-credentials/).
